gobblin-utility/src/main/java/org/apache/gobblin/util/logs/LogCopier.java

```diff
@@ -536,14 +521,15 @@ private void copyChangesOfLogFile(Path srcFile, Path destFile) throws IOExceptio
 
       try (Closer closer = Closer.create()) {
         fsDataInputStream = closer.register(LogCopier.this.srcFs.open(srcFile));
-        // Seek to the the most recent position if it is available
-        LOGGER.debug(String.format("Reading log file %s from position %d", srcFile, this.currentPos));
-        fsDataInputStream.seek(this.currentPos);
         BufferedReader srcLogFileReader = closer.register(
             new BufferedReader(new InputStreamReader(fsDataInputStream, ConfigurationKeys.DEFAULT_CHARSET_ENCODING)));
 
-        FSDataOutputStream outputStream = LogCopier.this.destFs.exists(destFile)
-            ? LogCopier.this.destFs.append(destFile) : LogCopier.this.destFs.create(destFile);
+        //If file already exists, delete and re-copy the file.
+        if (LogCopier.this.destFs.exists(destFile)) {
```

Comment:
Why do we need to re-copy the file ?

Reply:
Agreed. This is un-necessary. This condition will never be satisfied since we filter out files which have already been copied.